tests/unit/test_textUtils.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: UTF-8 -*-
+#tests/unit/test_textUtils.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2019 NV Access Limited, Babbage B.V., Leonard de Ruijter
+
+"""Unit tests for the textUtils module."""
+
+import unittest
+from textUtils import WideStringOffsetConverter
+
+class TestStrToWideOffsets(unittest.TestCase):
+	"""
+	Tests that ensure that offsets in a string are properly converted to wide string offsets.
+	Every string offset for 32-bit unicode characters (e.g. emoji) take two offsets in a wide string representation.
+	"""
+
+	def test_nonSurrogate(self):
+		converter = WideStringOffsetConverter(text="abc")
+		self.assertEqual(converter.wideStringLength, 3)
+		self.assertEqual(converter.strToWideOffsets(0, 0), (0, 0))
+		self.assertEqual(converter.strToWideOffsets(0, 1), (0, 1))
+		self.assertEqual(converter.strToWideOffsets(0, 2), (0, 2))
+		self.assertEqual(converter.strToWideOffsets(0, 3), (0, 3))
+		self.assertEqual(converter.strToWideOffsets(1, 1), (1, 1))
+		self.assertEqual(converter.strToWideOffsets(1, 2), (1, 2))
+		self.assertEqual(converter.strToWideOffsets(1, 3), (1, 3))
+		self.assertEqual(converter.strToWideOffsets(2, 2), (2, 2))
+		self.assertEqual(converter.strToWideOffsets(2, 3), (2, 3))
+		self.assertEqual(converter.strToWideOffsets(3, 3), (3, 3))
+
+	def test_surrogatePairs(self):
+		converter = WideStringOffsetConverter(text=u"\U0001f926\U0001f60a\U0001f44d") # 🤦😊👍
```

Comment:
Since there are several characters being combined in different ways, it might be clearer to specify them as constants and join them together in their various combinations for each test:

```
facePalm = u"\U0001f926"  # 🤦
smile = u"U0001f60a" # 😊
thumbsUp = u"\U0001f44d" # 👍

...
converter = WideStringOffsetConverter(text=u"".join([facePalm, smile, thumbsUp]))
```

Reply:
I think this would make things a lot more readable here. Rather than using `join` as I suggested before, just add them together, the difference in performance will be negligible. Unless there is something you are trying to convey by having the unicode values there? I get that there is some benefit to seeing the number of bytes for each character, perhaps there is another way to achieve that? At the moment its very hard to tell that it is the same value being used throughout.

```
facePalm = u"\U0001f926"  # 🤦
smile = u"U0001f60a" # 😊
thumbsUp = u"\U0001f44d" # 👍

...
converter = WideStringOffsetConverter(text=facePalm + smile + thumbsUp)
```